Makefile

```diff
@@ -27,24 +27,45 @@ all: check \
 dist/pyodide_py.tar: $(wildcard src/py/pyodide/*.py)  $(wildcard src/py/_pyodide/*.py)
 	cd src/py && tar --exclude '*__pycache__*' -cf ../../dist/pyodide_py.tar pyodide _pyodide
 
-dist/pyodide.asm.js: \
+src/core/pyodide_pre.o: src/js/_pyodide.out.js src/core/pre.js
+	rm -f tmp.dat
+	echo '()<::>{' >> tmp.dat
+	cat src/js/_pyodide.out.js >> tmp.dat
+	echo '}' >> tmp.dat
+	cat src/core/pre.js >> tmp.dat
+	echo "pyodide_js_init();" >> tmp.dat
+
+	rm -f src/core/pyodide_pre.gen.c
+	echo '__attribute__((used)) __attribute__((section("em_js"), aligned(1)))' >> src/core/pyodide_pre.gen.c
+	echo 'char __em_js__pyodide_js_init[] = {'  >> src/core/pyodide_pre.gen.c
+	cat tmp.dat  | xxd -i - >> src/core/pyodide_pre.gen.c
+	echo ', 0};' >> src/core/pyodide_pre.gen.c
+
+	rm tmp.dat
+	emcc -c src/core/pyodide_pre.gen.c -o src/core/pyodide_pre.o
+
+dist/libpyodide.a: \
 	src/core/docstring.o \
 	src/core/error_handling.o \
 	src/core/hiwire.o \
 	src/core/_pyodide_core.o \
 	src/core/js2python.o \
 	src/core/jsproxy.o \
-	src/core/main.o  \
 	src/core/pyproxy.o \
 	src/core/python2js_buffer.o \
 	src/core/python2js.o \
-	src/js/_pyodide.out.js \
+	src/core/pyodide_pre.o
+	emar rcs dist/libpyodide.a $(filter %.o,$^)
```

Comment:
We won't be able to server .a files from the CDN without whitelisting the extension (which requires a manual request). Does the extension matters much here: A libpyodide.wasm or libpyodide.data wouldn't be correct?

From a MIME type point of view,it would be [`application/octet-stream`](https://mimetype.io/application/octet-stream)  or `application/wasm` anyway as far as I understand.

Reply:
I'm not sure: if you say `-lpyodide` and you have a file called `libpyodide.data` it will probably be linked? Note that `.a` files are archives though and not wasm files.

Reply:
I think these would be distributed more from github releases than from the cdn though.